compiler/src/main/java/org/qbicc/graph/CmpAndSwap.java

```diff
@@ -113,12 +146,34 @@ public static CompoundType getResultType(CompilationContext ctxt, ValueType valu
         CompoundType compoundType = map.get(valueType);
         if (compoundType == null) {
             TypeSystem ts = ctxt.getTypeSystem();
-            compoundType = ts.getCompoundType(CompoundType.Tag.NONE, null, -1, -1, () -> List.of(ts.getCompoundTypeMember(null, valueType, -1, -1), ts.getCompoundTypeMember(null, ts.getBooleanType(), -1, -1)));
+            compoundType = CompoundType.builder(ts)
+                .setTag(CompoundType.Tag.NONE)
```

Comment:
How does this connect to the new `CompoundType.TAG.INLINE` added in #620?

Reply:
This line would be replaced with creating an INLINE compound type. Probably makes sense to remove the change from this pull request all together.

Reply:
I've removed this change since it will be correctly modified in #620